pennylane/fourier/visualization.py

```diff
@@ -0,0 +1,692 @@
+            f"Fourier coefficients. Received input of type {type(coeffs)}"
+        )
+
+    # In case we have a list, turn it into a numpy array
+    if isinstance(coeffs, list):
+        coeffs = np.array(coeffs)
+
+    # Check if the user provided a single set of coefficients to a function that is
+    # meant to accept multiple samples; add an extra dimension around it if needed
+    if len(coeffs.shape) == n_inputs and can_be_list:
+        coeffs = np.array([coeffs])
+
+    # Check now that we have the right number of axes for the type of function
+    required_shape_size = n_inputs + 1 if can_be_list else n_inputs
+    if len(coeffs.shape) != required_shape_size:
+        raise ValueError(
+            f"Plotting function expected a list of {n_inputs}-dimensional inputs. "
+            f"Received coefficients of {len(coeffs.shape)}-dimensional function."
+        )
+
+    # Shape in all dimensions of a single set of coefficients must be the same
+    shape_set = set(coeffs.shape[1:]) if can_be_list else set(coeffs.shape)
+    if len(shape_set) != 1:
+        raise ValueError(
+            "All dimensions of coefficient array must be the same. "
+            f"Received array with dimensions {coeffs.shape}"
+        )
+
+    # Size of each sample dimension must be 2d + 1 where d is the degree
+    shape_dim = coeffs.shape[1] if can_be_list else coeffs.shape[0]
+    if (shape_dim - 1) % 2 != 0:
+        raise ValueError(
+            "Shape of input coefficients must be 2d + 1, where d is the largest frequency. "
+            f"Coefficient array with shape {coeffs.shape} is invalid."
+        )
+
+    # Return the coefficients; we may have switched to a numpy array or added a needed extra dimension
+    return coeffs
+
+
+def _extract_data_and_labels(coeffs):
+    """Helper function for creating frequency labels and partitionining data.
+
+    Args:
+        coeffs (array[complex]): A list of sets of Fourier coefficients.
+
+    Returns:
+        (list(str), dict[str, array[complex]): The set of frequency labels, and a data
+            dictionary split into real and imaginary parts.
+    """
+    # extract the x ticks
+    nvecs = list(to_dict(coeffs[0]).keys())
+    nvecs_formatted = [format_nvec(nvec) for nvec in nvecs]
+
+    # make data
+    data = {}
+    data["real"] = np.array([[c[nvec].real for nvec in nvecs] for c in coeffs])
+    data["imag"] = np.array([[c[nvec].imag for nvec in nvecs] for c in coeffs])
+
+    return nvecs_formatted, data
+
+
+def _adjust_spine_placement(ax):
+    """Helper function to set some common axis properties when plotting."""
+    ax.xaxis.grid()
+    ax.spines["top"].set_visible(False)
+    ax.spines["bottom"].set_position("zero")
+    ax.spines["right"].set_visible(False)
+    ax.set_axisbelow(True)
+
+
+def coefficients_violin_plot(coeffs, n_inputs, ax, colour_dict=None, show_freqs=True):
+    """Plots a list of sets of Fourier coefficients as a violin plot.
```

Comment:
@glassnotes I would definitely consider copying over violin-plot-specific details from the API page to this docstring, as this could easily be a landing page for a lot of users (and the module docstring is great!)

Reply:
Have added an example, including a graphic, to each of these plotting functions. It's the same example each time and so it's more copy/pasting than I would like, let me know if you think it would be better to use a shorter example.